Meetups/25/2025-03-14-minutes.md

```diff
@@ -0,0 +1,181 @@
+
+Robert: In spring boot you can write an agent using annotations. As dev I don't care about the protocol. The TD is generated by the protocol. It's a code first framework.
+
+Robert: we are using Thingweb for runtime and Webthings for gateway control. I converted the things in the testing Webthings gateway to Agent Tools. I can ask a questions and see the result of the actions.
+
+> Shows a demo
+
+Robert: I just used the description and I was able to create a scenario. We open-sourced the whole LMOS platform.
+
+Robert: We defined Capabilities to augment Agent description in TD. Operator can fetch Agent Capabilities. We have build a complex CI/CD flow specific for agents, we support canary deployments.
+
+Robert: We defined the concept of channels to route the right query to the right agent. The entrypoint is Supervisor Agent that talks with LMOS operator which is capable of creating channel information.
+
+> Shows a demo
+
+Robert: if you don't have the agent capable of handling a request the query the system respond accordingly, but you can later install it.
+
+Robert: Additionally, the system automatically provide different context information: user and system.
+
+Robert: the whole project is a WIP let's collaborate to make it real.
+
+### Discussion
+
+Ege: thank you for the presentation and thank you for making it open source.
+
+Samuele: Are the use cases part of the standard or custom for agents.
+
+Robert: They are not standard. They are made for Arc (Agent DSL).
+
+Samuele: It reminded my MCP for prompts.
+
+Ben: Really cool project, great to see to use  Web Thing protocol. I'm curious about agents interacting with devices using TDs. The interesting part is Agent talk with each other, in particular using natural language. I show a demo using a particular audio protocol (tones instead of natural language).
+
+Robert: Quite interesting demo, we are looking in M to M too. There might be more performant protocol than audio. The protocol negotiation is very important. We can re-use existing solutions and I like the meta idea of WoT, protocol and media type negotiation is very interesting. P2P for WoT is also interesting. There is a new agency formed to build a new protocol... I fear that we are re-inventing the wheel. Is important we collaborate and build something together.
+
+Samuele: We worked on context switching <https://arxiv.org/abs/2410.11905>
+
+Sam: Is there some platform to publish agents? Like pip or npm?
+
+Robert: In MCP there are multiple registries. We will built something similar on top of WoT. We can start working on it in DT, but later make it public. Another interesting point is micro transactions between agents.
+
+Jonathan: How to know which is the best agents when getting a query?
+
+Robert: It is challenging.
+
+Jonathan: Could we end up to have something similar to SEO?
+
+Robert: That's good point, it might be a possible scenario.
+
+Jonathan: Does it make sense to use metadata to pick the right agent.
```

Comment:
```suggestion
Jonathan: Could we end up having something similar to SEO?

Robert: That's good point. It might be a possible scenario.

Jonathan: Does it make sense to use metadata to pick the right agent?
```